sw/device/lib/base/crc32.c

```diff
@@ -10,26 +10,37 @@
 #include "sw/device/lib/base/memory.h"
 
 #ifdef OT_PLATFORM_RV32
+enum {
+  /**
+   * CRC32 polynomial.
+   */
+  kCrc32Poly = 0xedb88320,
+  kCrc32Mu = 0xf7011641,
+};
+
 OT_WARN_UNUSED_RESULT
-static uint32_t crc32_internal_add8(uint32_t ctx, uint8_t byte) {
-  ctx ^= byte;
+static uint32_t crc32_i(uint32_t x) {
+  uint32_t result;
   asm(".option push;"
-      ".option arch, +zbr0p93;"
-      "crc32.b %0, %1;"
+      ".option arch, +zbc;"
```

Comment:
The `-march` arch string we use already includes zbc, so the `.option arch, +zbc` shouldn't be needed here.